src/core/execute-serialize.c

```diff
@@ -2720,10 +2720,9 @@ static int exec_context_deserialize(ExecContext *c, FILE *f) {
                         if (r < 0)
                                 return r;
                 } else if ((val = startswith(l, "exec-context-private-tmp="))) {
-                        r = parse_boolean(val);
-                        if (r < 0)
-                                return r;
-                        c->private_tmp = r;
+                        c->private_tmp = private_tmp_from_string(val);
```

Comment:
Previously, the field was serialized with `serialize_bool_elide()`. Hence, `"yes"` needs to be handled here for backward compatibility.

Reply:
For execute-serialize we don't maintain serialization format, but instead pin the executor to make sure compatible version is used, no?

Reply:
Please check out the comments in execute-serialization.h.